searxng/searxng
# Theme: implement search_on_category_select as a preference

## What does this PR do?

The implementation is change to work without javascript:
When enabled, the HTML page contains buttons instead of checkboxes.

## Why is this change important?

SearXNG have two modes to pick the categories:
* one as other search engine: when clicked, it sends the user query
* multiple categories: the user can pick multiple categories and then send the query.

The HTML always use checkboxes and labels --> some Javascript send the query in the first mode when the user clicks on a category.

This creates to UX issues:
* screen reader and text browser read or show the categories as check boxes.
* users who disable Javascript can't use the first mode.

This PR intends to fix that.

## How to test this PR locally?

* `make static.build.commit`
* `make run` : check in the multiple screen configuration (RTL, dark mode, mobile, tablet, desktop).

## Author's checklist

This is WIP because I would like to give some time to try to factorize the CSS.

## Related issues

* Related to https://github.com/searxng/searxng/issues/769#issuecomment-1132481995
* Fix https://github.com/searxng/searxng/issues/854
